lib/data/external/AwsClient.js

```diff
@@ -24,6 +24,37 @@ class AwsClient {
         this._serverSideEncryption = config.serverSideEncryption;
         this._supportsVersioning = config.supportsVersioning;
         this._client = new AWS.S3(this._s3Params);
+        this._logger = new werelogs.Logger('AwsClient');
+    }
+
+    setup(cb) {
+        this._client.getBucketLocation({ Bucket: this._awsBucketName },
+        (err, res) => {
+            if (err && err.code !== 'AuthorizationHeaderMalformed') {
```

Comment:
did you check the null ?

Reply:
You mean if err is null? That condition should be taken care of.